### ISSUE TITLE
pkgsCross.spidermonkey_78: error[E0463]: can't find crate for `std` in rust hello world check

### ISSUE THREAD

**Issue by Atemu** [NixOS/nixpkgs#106805]
**Describe the bug**

Fails with

```
DEBUG: Creating `/build/conftest31nhza6s.rs` with content:
DEBUG: | pub extern fn hello() { println!("Hello world"); }
DEBUG: Executing: `/nix/store/wgy1kkbzi0ccm5zavxsnf9rhrhzlv30h-rustc-1.48.0/bin/rustc --crate-type staticlib --target=x86_64-unknown-linux-gnu -o /build/conftest1tt_fg6h.rlib /build/conftest31nhza6s.rs`
DEBUG: The command returned non-zero exit status 1.
DEBUG: Its error output was:
DEBUG: | error[E0463]: can't find crate for `std`
DEBUG: |
DEBUG: | error: aborting due to previous error
DEBUG: |
DEBUG: | For more information about this error, try `rustc --explain E0463`.
ERROR: Cannot compile for x86_64-unknown-linux-gnu with /nix/store/wgy1kkbzi0ccm5zavxsnf9rhrhzlv30h-rustc-1.48.0/bin/rustc
The target may be unsupported, or you may not have
a rust std library for that target installed. Try:

  rustup target add x86_64-unknown-linux-gnu
```

**To Reproduce**
`nix-build -A pkgsCross.armv7l-hf-multiplatform.spidermonkey_78`

**Additional context**

https://github.com/NixOS/nixpkgs/issues/106759

**Notify maintainers**

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
# a list of nixos modules affected by the problem
module:
```

**Comment by Atemu** [comment]
This specific bug does not happen when 9eb79c42eb8b6ea07add51d972de0df1d7383f71 is reverted, cc @petabyteboy. 

Any pointers?

We now have 

```
In file included from Unified_cpp_js_src_wasm2.cpp:38:
/build/firefox-78.4.0/js/src/wasm/WasmSignalHandlers.cpp:540:19: error: field 'ufp' has incomplete type 'user_vfp'
  540 |   struct user_vfp ufp;
      |                   ^~~
/build/firefox-78.4.0/js/src/wasm/WasmSignalHandlers.cpp:540:10: note: forward declaration of 'struct user_vfp'
  540 |   struct user_vfp ufp;
      |          ^~~~~~~~
/build/firefox-78.4.0/js/src/wasm/WasmSignalHandlers.cpp:541:23: error: field 'ufp_exc' has incomplete type 'user_vfp_exc'
  541 |   struct user_vfp_exc ufp_exc;
      |                       ^~~~~~~
/build/firefox-78.4.0/js/src/wasm/WasmSignalHandlers.cpp:541:10: note: forward declaration of 'struct user_vfp_exc'
  541 |   struct user_vfp_exc ufp_exc;
      |          ^~~~~~~~~~~~
```

causing a build failure, hidden in a sea of warnings but it's probably unrelated.

**Comment by petabyteboy** [comment]
What nixpkgs commit are you using exactly? This should have been fixed with #106790.

**Comment by Atemu** [comment]
None new enough to have that PR in it, thank you!
